e2e/e2eutil/node.go

```diff
@@ -0,0 +1,94 @@
+package e2eutil
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/helper/uuid"
+)
+
+// NodeRestart is a test helper function that restarts a client node
+// running under systemd using a raw_exec job. Returns the job ID of
+// the restart job so that callers can clean it up.
+func NodeRestart(client *api.Client, nodeID string) (string, error) {
+	ok, err := isUbuntu(client, nodeID)
+	if !ok {
+		// TODO(tgross): we're checking this because we want to use
+		// systemctl to restart the node, but we should also figure
+		// out a way to detect dev mode targets.
+		return "", fmt.Errorf("NodeRestart only works against ubuntu targets")
+	}
+	if err != nil {
+		return "", err
+	}
+
+	job := newRestartJob(nodeID)
+	jobID := *job.ID
+	_, _, err = client.Jobs().Register(job, nil)
+	if err != nil {
+		return jobID, err
+	}
+
+	retries := 30
+	for retries > 0 {
+		time.Sleep(1 * time.Second)
```

Comment:
This does not seem like a sufficient to assert the node restarted. Does it seem to work when you manually monitor it?

Perhaps doing a blocking restart with 0 job retries and checking the alloc status would be better.

Reply:
It does seem to work if I manually monitor it, but you're right... I might be getting "lucky" and if restarts start taking longer in the future it'd give us false confidence in the result. Will fix.

Reply:
Done. I changed the restart to blocking, then I poll the alloc until it's marked `dead`, and only then start checking the node status for readiness.